File: mifosng-android/src/main/java/com/mifos/objects/client/ActivatePayload.java
package com.mifos.objects.client;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * Created by Rajan Maurya on 09/02/17.
 */

public class ActivatePayload implements Parcelable {

    @SerializedName("activationDate")
    String activationDate;

    @SerializedName("dateFormat")
    String dateFormat = "dd MMMM YYYY";

    @SerializedName("locale")
    String locale = "en";

    public String getActivationDate() {
        return activationDate;
    }

    public void setActivationDate(String activationDate) {
        this.activationDate = activationDate;
    }

    public String getDateFormat() {
        return dateFormat;
    }

    public void setDateFormat(String dateFormat) {
        this.dateFormat = dateFormat;
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(String locale) {
        this.locale = locale;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.activationDate);
        dest.writeString(this.dateFormat);
        dest.writeString(this.locale);
    }

    public ActivatePayload(String activationDate) {
        this.activationDate = activationDate;
    }

    public ActivatePayload() {
    }

    protected ActivatePayload(Parcel in) {
        this.activationDate = in.readString();
        this.dateFormat = in.readString();
        this.locale = in.readString();
    }

    public static final Parcelable.Creator<ActivatePayload> CREATOR = new Parcelable
            .Creator<ActivatePayload>() {
        @Override
        public ActivatePayload createFromParcel(Parcel source) {
            return new ActivatePayload(source);
        }

        @Override
        public ActivatePayload[] newArray(int size) {
            return new ActivatePayload[size];
        }
    };
}

File: core/model/src/commonMain/kotlin/com/mifos/core/model/objects/clients/ActivatePayload.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.model.objects.clients

import com.mifos.core.common.utils.Parcelable
import com.mifos.core.common.utils.Parcelize

/**
 * Created by Rajan Maurya on 09/02/17.
 */
@Parcelize
data class ActivatePayload(
    var activationDate: String? = null,

    var dateFormat: String? = "dd MMMM YYYY",

    var locale: String? = "en",
) : Parcelable
